modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -387,8 +387,8 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
    * Determine an address' type based on its witness and redeem script presence
    * @param addressDetails
    */
-  static inferAddressType(addressDetails: { chain: number }): ScriptType2Of3 | null {
-    return isChainCode(addressDetails.chain) ? scriptTypeForChain(addressDetails.chain) : null;
+  static inferAddressType(addressDetails: { chain?: number }): ScriptType2Of3 | null {
+    return isChainCode(addressDetails?.chain) ? scriptTypeForChain(addressDetails?.chain) : null;
```

Comment:
unclear why you are doing this change

Reply:
Because of this https://github.com/BitGo/BitGoJS/pull/3440#discussion_r1156879526 

It was throwing before since it did no allow undefined. With this, undefined is accepted.

Reply:
But then you would also have to change the type signature here.

But instead of allowing `undefined` here rather just handle it at the callsite.